runtime/reconcilers/model.go

```diff
@@ -1030,32 +1031,51 @@ func (r *ModelReconciler) executeAll(ctx context.Context, self *runtimev1.Resour
 	defer release()
 
 	// First step is to resolve and sync the partitions.
-	err = r.resolveAndSyncPartitions(ctx, self, model, incrementalState)
-	if err != nil {
-		return "", nil, false, fmt.Errorf("failed to sync partitions: %w", err)
+	// Skip this if we're doing an explicit partition refresh (TriggerPartitions), as we only want to execute the explicitly triggered partitions
+	if !model.Spec.TriggerPartitions {
+		err = r.resolveAndSyncPartitions(ctx, self, model, incrementalState)
+		if err != nil {
+			return "", nil, false, fmt.Errorf("failed to sync partitions: %w", err)
+		}
 	}
 
 	// Track execution metadata
 	var totalExecDuration atomic.Int64
 	firstRunIsIncremental := incrementalRun
 
+	var explicitlyTriggeredPartitions []drivers.ModelPartition
+	explicitlyTargetedRefresh := model.Spec.TriggerPartitions
+	partitionFilter := &drivers.FindModelPartitionsOptions{
+		ModelID: model.State.PartitionsModelId,
+	}
+
+	// When partitions are explicitly triggered, process both explicitly triggered partitions AND any pending partitions in the same run
+	partitionFilter.WherePending = true
+	partitionFilter.WhereExplicitlyTriggered = model.Spec.TriggerPartitions
```

Comment:
Looking over this, why does it even need the `explicitly_triggered` field and related queries/checks? It should never _not_ run pending partitions. It just should not call `resolveAndSyncPartitions` (which discovers new partitions) when only a specific partition has been triggered (i.e. marked pending).